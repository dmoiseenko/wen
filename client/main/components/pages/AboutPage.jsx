import React from 'react';
import Markdown from 'react-remarkable';
import renderHTML from 'react-render-html';


const innerHtml = `
  <h5>
    Created by <a href="mailto:dmoiseenko@mykolab.com">Denis Moiseenko</a>
  </h5>
`;

const markdownToHtml = `
  ##### All source code is available on [here](https://moiden.com/dmoiseenko/wen)
`;

const htmlToReactComponents = `
  <h5>
     <a href="https://moiden.com/dmoiseenko/wen/blob/master/LICENSE">MIT License</a>
  </h5>
`;

export default () => (
  <div className="container">
    <div className="content">
      <h2>About</h2>
      <br />

      <div dangerouslySetInnerHTML={{ __html: innerHtml }} />

      <Markdown>
        {markdownToHtml}
      </Markdown>

      {renderHTML(htmlToReactComponents)}
    </div>
  </div>
);
