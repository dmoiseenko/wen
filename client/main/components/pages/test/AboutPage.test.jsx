import React from 'react';
import { shallow } from 'enzyme';

import AboutPage from '../AboutPage';


it('should render', () => {
  const wrapper = shallow(
    <AboutPage />
  );

  expect(wrapper.node).toMatchSnapshot();
});
