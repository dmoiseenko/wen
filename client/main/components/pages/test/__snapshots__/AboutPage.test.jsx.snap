// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <h2>
      About
    </h2>
    <br />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "
          <h5>
            Created by <a href=\\"mailto:dmoiseenko@mykolab.com\\">Denis Moiseenko</a>
          </h5>
        ",
        }
      }
    />
    <Remarkable
      container="div"
      options={Object {}}
    >
      
        ##### All source code is available on [here](https://moiden.com/dmoiseenko/wen)
      
    </Remarkable>
    
      
    <h5>
      
           
      <a
        href="https://moiden.com/dmoiseenko/wen/blob/master/LICENSE"
      >
        MIT License
      </a>
      
        
    </h5>
    
    
  </div>
</div>
`;
